### ISSUE TITLE
Rutracker_check plugin (nnmclub.to)

### ISSUE THREAD

**Issue by IvanShift** [Novik/ruTorrent#2307]
### Please complete the following tasks.

- [X] Web browser cache cleared
- [X] Copied everything if upgrading to v4.0 or master
- [X] Web browser, ruTorrent and PHP version provided

### Tell us about your environment

Web browser: Firefox
ruTorrent: 3.10

### Give us detailed steps to reproduce the error

Hello! 
A couple of months ago, torrents with nnmclub stopped auto-updating (Error accessing the tracker).
I suppose that can be connected with nnm-club.ws. This mirror practically stops working. 
Could you please check it?

### Tell us the error message

Error accessing the tracker

**Comment by stickz** [comment]
Could you take a screenshot of the trackers tab on one of these torrents? And attach it to this issue report?

**Comment by IvanShift** [comment]
![изображение](https://user-images.githubusercontent.com/64527940/154441581-6dd02fcb-cef1-4d0a-8d60-cc7fd464c373.png)

This torrent:
https://nnmclub.to/forum/viewtopic.php?p=11178274

**Comment by stickz** [comment]
The reason why it's doing that is because the trackers are incorrect. How was this torrent added to ruTorrent?

**Comment by IvanShift** [comment]
Everything is correct. You can check it yourself.
[[NNMClub.to]_SDI_Update.zip](https://github.com/Novik/ruTorrent/files/8087407/NNMClub.to._SDI_Update.zip)

**Comment by stickz** [comment]
I am able to access the `nnmclub.cc` tracker on both of these torrent files. I don't think this is a problem with ruTorrent.
![tracker_result](https://user-images.githubusercontent.com/9171157/154450326-87bbd0ba-9107-4647-a594-fff5875480fd.PNG)

**Comment by IvanShift** [comment]
I was never told that I cannot download this torrent. 
I mean that I can't update it via ruTorrent. Try right-clicking on that torrent and select "Check for Updates".  After that I see this:
![изображение](https://user-images.githubusercontent.com/64527940/154451535-a77a3190-b285-437b-b3b3-284e141e75a0.png)

**Comment by Novik** [comment]
@IvanShift 
Please, open file ruTorrent/plugins/rutracker_check/trackers/nnmclub.php
and change fragment
```
$client->fetchComplex("https://nnm-club.ws/download.php?id=".$matches["id"]);
```
to
```
$client->fetchComplex("https://nnmclub.to/forum/download.php?id=".$matches["id"]);
```
Does it work now?

**Comment by IvanShift** [comment]
As I said simply changing to nnmclub.to doesn't help.

**Comment by Novik** [comment]
Do you have correct credentials for nnmclub in the loginmgr plugin?

**Comment by n1ce-0ne** [comment]
just tested it on my install to try help out, seems to update ok here & no errors, http://bt02.nnm-club.cc:2710.......... worked fine while right clicking & updating trackers, ive no other info added or changed like any credentials/SID/urls

**Comment by IvanShift** [comment]
@n1ce-0ne Everything is fine with downloading and trackers. The problem is only with the plugin which provides auto-update torrent-file after it's updated on the tracker.

**Comment by n1ce-0ne** [comment]
ahhh cool ok, thanks for letting me know.

**Comment by Novik** [comment]
As result, what we have here.
This is a russian-specific RKN trouble.
Replacement above (https://github.com/Novik/ruTorrent/commit/0d85f1e970176e6e44a5bec98d18fe2e56eca83d) and usage of HTTPs proxy is enough for fixing it.

> I suppose that it is necessary to add SID-parameter to this request.

This processed on the loginmgr plugin.

@IvanShift 

Ходите в config.php и настраивайте HTTPS прокси. Либо используйте любой другой способ обхода блокировок, на том же nnmclub их предостаточно перечислено.
Видимо, в последние два месяца ваш интернет провайдер наконец озаботился корректной блокировкой веб-морды данного трекера.

**Comment by IvanShift** [comment]
@Novik 
Да, вы правы, дело в блокировке. Может быть у вас есть какие-то на примете https-прокси или сбособы обхода, которые можно использовать на роутере? Пока не получается найти что-то рабочее.

**Comment by IvanShift** [comment]
@Novik 
I found this one: HTTPS [proxy-ssl.antizapret.prostovpn.org:3143](http://proxy-ssl.antizapret.prostovpn.org:3143/)
After that, I have: "Result:	No need". But it's not true. I see an updated tracker.

**Comment by Novik** [comment]
This message means: 
Comment in your .torrent file does not satisfy to the PCRE expression ```^https?://(nnm-club|nnmclub)\.(ru|me|to|name|tv)/forum/viewtopic\.php\?p=(?P<id>\d+)$```

**Comment by IvanShift** [comment]
Hello! Everything is fine now! Thank you!
Is it possible to use HTTPS proxy only for selected hosts? Can you add this feature?

**Comment by Novik** [comment]
> Is it possible to use HTTPS proxy only for selected hosts?

This is a good point, I will think about that.
